lib/schema_plus/core/active_record/schema_dumper.rb

```diff
@@ -66,17 +66,22 @@ def table(table, _)
               env.table.pname = m[:name]
               env.table.options = m[:options].strip
               env.table.trailer = m[:trailer].split("\n").map(&:strip).reject{|s| s.blank?}
-              env.table.columns = m[:columns].strip.split("\n").map { |col|
-                m = col.strip.match %r{
+              m[:statements].strip.split("\n").each do |statement|
+                statement.strip!
+                m = statement.match %r{
                 ^
                 t\.(?<type>\S+) \s*
                   [:'"](?<name>[^"\s]+)[,"]? \s*
                   ,? \s*
                   (?<options>.*)
                 $
                 }x
-                SchemaDump::Table::Column.new name: m[:name], type: m[:type], options: eval("{" + m[:options] + "}"), comments: []
-              }
+                if m.nil?
+                  env.table.statements << statement
```

Comment:
Looks like the `t.index` statement will actually need to be parsed and turned into a `SchemaDump::Table::Index` instance (or something similar that outputs a new `t.index` string). Not sure whether it makes sense to add it to `env.table.statements` or have something like `env.table.indexes`.